openedx_learning/core/publishing/api.py

```diff
@@ -107,6 +162,37 @@ def learning_package_exists(key: str) -> bool:
     return LearningPackage.objects.filter(key=key).exists()
 
 
+def get_last_publish(learning_package_id: int) -> PublishLog | None:
+    return PublishLog.objects \
+                     .filter(learning_package_id=learning_package_id) \
+                     .order_by('-id') \
+                     .first()
+
+def get_all_drafts(learning_package_id: int):
+    return Draft.objects.filter(
+        entity__learning_package_id=learning_package_id,
+        version__isnull=False,
+    )
+
+def get_entities_with_unpublished_changes(learning_package_id: int):
+    return PublishableEntity.objects \
+               .filter(learning_package_id=learning_package_id) \
+               .exclude(draft__version=F('published__version'))
+
+def get_entities_with_unpublished_deletes(learning_package_id: int):
```

Comment:
What's the intended use case of this API method? Perhaps a widget in the UI that says something like "These things have been deleted since your last publish: ____" ?

Reply:
I'm not entirely sure, actually. It's returned as [part of the metadata in the `get_library` call](https://github.com/openedx/edx-platform/blob/5838d68efc1bd29d9dda593d1d0a1306654e58ba/openedx/core/djangoapps/content_libraries/api.py#L393-L394).

Reply:
In the [new code](https://github.com/openedx/edx-platform/pull/34066/files#diff-a760c3e87f2f671ac2453abbbaf6f10ca052c7acf1c46c1d36f24646e1d52a00R334-R339), I just run `.exists()` on the queryset returned here.